data-node/src/main/java/org/graylog/datanode/opensearch/statemachine/OpensearchStateMachine.java

```diff
@@ -108,7 +108,8 @@ public static OpensearchStateMachine createNew(OpensearchProcess process, Set<St
                 .permit(OpensearchEvent.HEALTH_CHECK_OK, OpensearchState.AVAILABLE)
                 .permit(OpensearchEvent.PROCESS_STOPPED, OpensearchState.TERMINATED)
                 .permit(OpensearchEvent.PROCESS_PREPARED, OpensearchState.PREPARED) //restart if reconfigured
-                .permit(OpensearchEvent.PROCESS_TERMINATED, OpensearchState.TERMINATED);
+                .permit(OpensearchEvent.PROCESS_TERMINATED, OpensearchState.TERMINATED)
+                .permit(OpensearchEvent.PROCESS_STARTED, OpensearchState.STARTING);
```

Comment:
why did you add the STARTED transition here? i don't see where it is triggered

Reply:
I encountered a situation where the datanode startup failed repeatedly due to invalid certificate. Then I started the datanode server, this generated an usable cert for the datanode. Datanode could consume the cert but failed to start, because transitions from terminated to starting were not allowed.